cloudmarker/test/test_filestore.py

```diff
@@ -0,0 +1,69 @@
+"""Tests for filestore plugins."""
+
+
+import os
+import pathlib
+import shutil
+import unittest
+
+from cloudmarker.stores import filestore
+
+
+class FileStoreTest(unittest.TestCase):
+    """Tests for filestore plugins."""
+
+    @classmethod
+    def write_to_file(cls, filename):
+        # Initialize the FileStore plugin with the filename
+        store = filestore.FileStore(filename)
+
+        # dummy data to write
+        mp = {
+            "record_type": "dummy",
+            "data": {
+                "test_name": "tilde_filename"
+            }
+        }
+
+        store.write(mp)
+        store.done()
+
+    def test_filepath_tilde(self):
+        self.write_to_file("~/caramel_l33t/cloudmarker")
+        # if the parsing fails, it should create a "~" directory in the current
+        # path
+        f = pathlib.Path("~")
+
+        # If exists and is a directory, the parsing failed.
+        self.assertFalse(f.exists() and f.is_dir())
+
+    def test_filepath_tilde2(self):
+        user = os.getenv("USER")
+        self.write_to_file("~{}/caramel_tpb/cloudmarker".format(user))
+        # if the parsing fails, it should create a "~" directory in the current
+        # path
+        f = pathlib.Path("~{}".format(user))
+
+        # If exists and is a directory, the parsing failed.
+        self.assertFalse(f.exists() and f.is_dir())
+
+    def test_filepath_dir_traversal(self):
+        self.write_to_file("/tmp/cloudmarker/../abc")
+
+        f = pathlib.Path("/tmp/abc")
+        self.assertTrue(f.exists() and f.is_dir())
+
+    @classmethod
+    def tearDownClass(cls):
+        # remove all junk files created
+        user = os.getenv("USER")
+        p = [
+            "/tmp/abc",
+            "~/caramel_l33t",
+            os.path.join("~{}".format(user), "caramel_l33t")
+        ]
+
+        for path in p:
+            if pathlib.Path(path).exists():
+                # delete the directory recursively
+                shutil.rmtree(path)
```

Comment:
This command did not delete the directories created for me.

Reply:
Looks like the `~` in the path needs to undergo tilde expansion so that the existence check succeeds and the folder is deleted. There is `pathlib.Path(path).expanduser()` but it was introduced in version 3.5, so using this would cause Travis CI builds to fail because we have Python 3.4 also in [the build matrix](https://github.com/cloudmarker/cloudmarker/blob/955d4ef/.travis.yml#L3-L5). So that leaves us with `os.path.exists(os.path.expanduser(path))`.

Reply:
I have added the following folder names, so now it should delete all folder with and without path expansion.
```python
user = os.getenv("USER")
p = [
    "/tmp/abc",
    os.path.expanduser("~/caramel_l33t"),
    os.path.expanduser("~/caramel_tpb"),
    "~{}".format(user),
    "~"
]
```